### PR TITLE
Massive reformat to fix whitespace inconsistencies

### PR DESCRIPTION
**Pull request by igitur**
cc @tournierjc 

My suggested PR to fix the whitespace problems. I also added a [CONTRIBUTING.md](https://github.com/igitur/qlnet/blob/massive-reformat-to-fix-whitespace-inconsistencies/CONTRIBUTING.md) file that explains the process to avoid any future whitespace problems.

This PR will create once-off pain, but will ease development in the future, provided that contributors stick to the guidelines.

### CODE REVIEW

**Comment by igitur** [comment]
I will fix the conflicts later, but let's discuss the concepts of this PR first.

File: src/QLNet/Currencies/America.cs [hunk 1]
```diff
@@ -2,149 +2,160 @@
...
+      public MXNCurrency() : base("Mexican peso", "MXN", 484, "Mex$", "", 100, new Rounding(), "%3% %1$.2f") { }
+   }
+
+   //! Peruvian nuevo sol
+   //    ! The ISO three-letter code is PEN; the numeric code is 604.
+   //        It is divided in 100 centimos.
+   //
+   //        \ingroup currencies
+   //
+   public class PENCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'PENCurrency' to match camel case naming rules, consider using 'PenCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 2]
```diff
@@ -2,149 +2,160 @@
...
+      public TTDCurrency() : base("Trinidad & Tobago dollar", "TTD", 780, "TT$", "", 100, new Rounding(), "%3% %1$.2f") { }
+   }
+
+   //! U.S. dollar
+   //    ! The ISO three-letter code is USD; the numeric code is 840.
+   //        It is divided in 100 cents.
+   //
+   //        \ingroup currencies
+   //
+   public class USDCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'USDCurrency' to match camel case naming rules, consider using 'UsdCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 3]
```diff
@@ -2,149 +2,160 @@
...
+      public BRLCurrency() : base("Brazilian real", "BRL", 986, "R$", "", 100, new Rounding(), "%3% %1$.2f") { }
+   }
+
+   //! Canadian dollar
+   //    ! The ISO three-letter code is CAD; the numeric code is 124.
+   //        It is divided into 100 cents.
+   //
+   //        \ingroup currencies
+   //
+   public class CADCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'CADCurrency' to match camel case naming rules, consider using 'CadCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 4]
```diff
@@ -2,149 +2,160 @@
...
+      public COPCurrency() : base("Colombian peso", "COP", 170, "Col$", "", 100, new Rounding(), "%3% %1$.2f") { }
+   }
+
+   //! Mexican peso
+   //    ! The ISO three-letter code is MXN; the numeric code is 484.
+   //        It is divided in 100 centavos.
+   //
+   //        \ingroup currencies
+   //
+   public class MXNCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'MXNCurrency' to match camel case naming rules, consider using 'MxnCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 5]
```diff
@@ -2,149 +2,160 @@
...
+      public PEHCurrency() : base("Peruvian sol", "PEH", 999, "S./", "", 100, new Rounding(), "%3% %1$.2f") { }
+   }
+
+   //! Trinidad & Tobago dollar
+   //    ! The ISO three-letter code is TTD; the numeric code is 780.
+   //        It is divided in 100 cents.
+   //
+   //        \ingroup currencies
+   //
+   public class TTDCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'TTDCurrency' to match camel case naming rules, consider using 'TtdCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 6]
```diff
@@ -2,149 +2,160 @@
...
+      public ARSCurrency() : base("Argentinian peso", "ARS", 32, "", "", 100, new Rounding(), "%2% %1$.2f") { }
+   }
+
+   //! Brazilian real
+   //    ! The ISO three-letter code is BRL; the numeric code is 986.
+   //        It is divided in 100 centavos.
+   //
+   //        \ingroup currencies
+   //
+   public class BRLCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'BRLCurrency' to match camel case naming rules, consider using 'BrlCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 7]
```diff
@@ -2,149 +2,160 @@
...
+      public CLPCurrency() : base("Chilean peso", "CLP", 152, "Ch$", "", 100, new Rounding(), "%3% %1$.0f") { }
+   }
+
+   //! Colombian peso
+   //    ! The ISO three-letter code is COP; the numeric code is 170.
+   //        It is divided in 100 centavos.
+   //
+   //        \ingroup currencies
+   //
+   public class COPCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'COPCurrency' to match camel case naming rules, consider using 'CopCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 8]
```diff
@@ -2,149 +2,160 @@
...
+   //! Peruvian sol
+   //    ! The ISO three-letter code was PEH. A numeric code is not available
+   //        as per ISO 3166-1, we assign 999 as a user-defined code.
+   //        It was divided in 100 centavos.
+   //
+   //        Obsoleted by the inti since February 1985.
+   //
+   //        \ingroup currencies
+   //
+   public class PEHCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'PEHCurrency' to match camel case naming rules, consider using 'PehCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 9]
```diff
@@ -2,149 +2,160 @@
...
+      public USDCurrency() : base("U.S. dollar", "USD", 840, "$", "\xA2", 100, new Rounding(), "%3% %1$.2f") { }
+   }
 
+   //! Venezuelan bolivar
+   //    ! The ISO three-letter code is VEB; the numeric code is 862.
+   //        It is divided in 100 centimos.
+   //
+   //        \ingroup currencies
+   //
+   public class VEBCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'VEBCurrency' to match camel case naming rules, consider using 'VebCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 10]
```diff
@@ -2,149 +2,160 @@
...
+   //! Peruvian inti
+   //    ! The ISO three-letter code was PEI.
+   //        It was divided in 100 centimos. A numeric code is not available
+   //        as per ISO 3166-1, we assign 998 as a user-defined code.
+   //
+   //        Obsoleted by the nuevo sol since July 1991.
+   //
+   //        \ingroup currencies
+   //
+   public class PEICurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'PEICurrency' to match camel case naming rules, consider using 'PeiCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Currencies/America.cs [hunk 11]
```diff
@@ -2,149 +2,160 @@
...
+      public CADCurrency() : base("Canadian dollar", "CAD", 124, "Can$", "", 100, new Rounding(), "%3% %1$.2f") { }
+   }
+
+   //! Chilean peso
+   //    ! The ISO three-letter code is CLP; the numeric code is 152.
+   //        It is divided in 100 centavos.
+   //
+   //        \ingroup currencies
+   //
+   public class CLPCurrency : Currency
```
Comment (amaggiulli) [inline src/QLNet/Currencies/America.cs]: MINOR Rename class 'CLPCurrency' to match camel case naming rules, consider using 'ClpCurrency'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Cashflows/OvernightIndexedCoupon.cs [hunk 12]
```diff
@@ -120,128 +123,135 @@ public class OvernightIndexedCoupon : FloatingRateCoupon
...
+
       //! value dates for the rates to be compounded
       public List<Date> valueDates() { return valueDates_; }
-   
+
       private List<Date> valueDates_, fixingDates_;
       private List<double> fixings_;
-      int n_;
-      List<double> dt_;
+      private int n_;
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/OvernightIndexedCoupon.cs]: MAJOR Make 'n_' 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2933)

File: src/QLNet/Cashflows/OvernightIndexedCoupon.cs [hunk 13]
```diff
@@ -120,128 +123,135 @@ public class OvernightIndexedCoupon : FloatingRateCoupon
...
+      private List<double> dt_;
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/OvernightIndexedCoupon.cs]: MAJOR Make 'dt_' 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2933)

File: src/QLNet/Cashflows/LinearTsrPricer.cs [hunk 14]
```diff
@@ -230,48 +238,51 @@ private double GsrG( Date d)
...
+
+         public override double value(double strike)
          {
             return section_.vega(strike) - targetVega_;
          }
-         
-         SmileSection section_;
-         double targetVega_;
+
+         private SmileSection section_;
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/LinearTsrPricer.cs]: MAJOR Make 'section_' 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2933)

File: src/QLNet/Cashflows/LinearTsrPricer.cs [hunk 15]
```diff
@@ -230,48 +238,51 @@ private double GsrG( Date d)
...
+         private double targetVega_;
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/LinearTsrPricer.cs]: MAJOR Make 'targetVega_' 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2933)

File: src/QLNet/Cashflows/ConundrumPricer.cs [hunk 16]
```diff
@@ -2,894 +2,1031 @@
...
+
+      public void setMeanReversion(Handle<Quote> meanReversion)
+      {
+         if (meanReversion_ != null)
+            meanReversion_.unregisterWith(update);
+         meanReversion_ = meanReversion;
+         if (meanReversion_ != null)
+            meanReversion_.registerWith(update);
+         update();
+
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: MAJOR Constructor has 9 parameters, which is greater than the 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS107)

File: src/QLNet/Cashflows/ConundrumPricer.cs [hunk 17]
```diff
@@ -2,894 +2,1031 @@
...
+            return Rs * functionZ(calibratedShift);
+         }
+
+         public override double firstDerivative(double Rs)
+         {
+            double calibratedShift = calibrationOfShift(Rs);
+            return functionZ(calibratedShift) + Rs * derZ_derX(calibratedShift) / derRs_derX(calibratedShift);
+         }
+
+         public override double secondDerivative(double Rs)
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: CRITICAL Rename parameter 'Rs' to 'x' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)

File: src/QLNet/Cashflows/ConundrumPricer.cs [hunk 18]
```diff
@@ -2,894 +2,1031 @@
...
-            Func<double, double> f_;
-            private List<double> abscissas = new List<double>();
-            private List<double> functionValues = new List<double>();
-
-            public Spy(Func<double, double> f) {
-                f_ = f;
+            discountRatio_ = swapPaymentDiscounts_.Last() / discountAtStart_;
+         }
+
+         public override double value(double Rs)
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: CRITICAL Rename parameter 'Rs' to 'x' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)

File: src/QLNet/Cashflows/ConundrumPricer.cs (hunk 16 above)
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: MAJOR Make 'f_' 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2933)

File: src/QLNet/Cashflows/ConundrumPricer.cs [hunk 19]
```diff
@@ -2,894 +2,1031 @@
...
+         {
+            double calibratedShift = calibrationOfShift(Rs);
+            return Rs * functionZ(calibratedShift);
+         }
+
+         public override double firstDerivative(double Rs)
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: CRITICAL Rename parameter 'Rs' to 'x' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)

File: src/QLNet/Cashflows/ConundrumPricer.cs (hunk 16 above)
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: CRITICAL Rename parameter 'x' to 'v' to match the interface declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)

File: src/QLNet/Cashflows/ConundrumPricer.cs [hunk 20]
```diff
@@ -2,894 +2,1031 @@
...
-
-                    result += temp1 - o_.discountAtStart_;
-                    derivative_ -= o_.shapedSwapPaymentTimes_.Last() * temp1;
-                    return result;
-                }
-
-                public override double derivative(double UnnamedParameter1) { return derivative_; }
-                public void setSwapRateValue(double x) { Rs_ = x; }
-                public GFunctionWithShifts gFunctionWithShifts() { return o_; }
+            public override double value(double x)
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/ConundrumPricer.cs]: CRITICAL Rename parameter 'x' to 'v' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)

File: src/QLNet/Cashflows/CashFlows.cs [hunk 21]
```diff
@@ -339,15 +345,14 @@ public override double value(double zSpread)
...
             double NPV = CashFlows.npv(leg_, curve_, includeSettlementDateFlows_, settlementDate_, npvDate_);
-                return npv_ - NPV;
+            return npv_ - NPV;
          }
-
-        
       }
-      class BPSCalculator : IAcyclicVisitor 
+
+      private class BPSCalculator : IAcyclicVisitor
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/CashFlows.cs]: MINOR Rename class 'BPSCalculator' to match camel case naming rules, consider using 'BpsCalculator'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

File: src/QLNet/Cashflows/CashFlows.cs [hunk 22]
```diff
@@ -264,55 +269,56 @@ class IrrFinder : ISolver1d
...
       }
-      class ZSpreadFinder : ISolver1d
+
+      private class ZSpreadFinder : ISolver1d
       {
          private Leg leg_;
          private double npv_;
          private SimpleQuote zSpread_;
-         ZeroSpreadedTermStructure curve_;
+         private ZeroSpreadedTermStructure curve_;
```
Comment (amaggiulli) [inline src/QLNet/Cashflows/CashFlows.cs]: MAJOR Make 'curve_' 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2933)

**Comment by amaggiulli** [comment]
SonarQube analysis reported 81 issues
* ![CRITICAL][CRITICAL] 41 critical
* ![MAJOR][MAJOR] 20 major
* ![MINOR][MINOR] 19 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ImpliedVolatility.cs#L74](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Instruments/ImpliedVolatility.cs#L74): Rename parameter 'x' to 'v' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [Makevanillaswap.cs#L243](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Instruments/Makevanillaswap.cs#L243): Refactor this method to reduce its Cognitive Complexity from 38 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [NormalDistribution.cs#L54](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Distributions/NormalDistribution.cs#L54): Rename parameter 'x' to 'v' to match the interface declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [NormalDistribution.cs#L95](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Distributions/NormalDistribution.cs#L95): Rename parameter 'z' to 'v' to match the interface declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [NormalDistribution.cs#L308](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Distributions/NormalDistribution.cs#L308): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [NormalDistribution.cs#L426](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Distributions/NormalDistribution.cs#L426): Rename parameter 'x' to 'v' to match the interface declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [NormalDistribution.cs#L540](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Distributions/NormalDistribution.cs#L540): Rename parameter 'x' to 'v' to match the interface declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [Linearinterpolation.cs#L48](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Interpolations/Linearinterpolation.cs#L48): Rename parameter 'x' to 'd' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [Linearinterpolation.cs#L55](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Interpolations/Linearinterpolation.cs#L55): Rename parameter 'x' to 'd' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)
1. ![CRITICAL][CRITICAL] [Linearinterpolation.cs#L62](https://github.com/amaggiulli/QLNet/blob/5d5b3f378263df4f0db3d442f3956c3ae2bb9230/src/QLNet/Math/Interpolations/Linearinterpolation.cs#L62): Rename parameter 'x' to 'd' to match the base class declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by amaggiulli** [comment]
No thank you , better step by step

**Pull request closed without merging**